thirdParty/cmake-modules/FindSplash.cmake

```diff
@@ -114,7 +114,7 @@ if(Splash_ROOT_DIR)
     #if(Splash_USE_STATIC_LIBS)
     #    set(HDF5_USE_STATIC_LIBRARIES ON)
     #endif()
-    find_package(HDF5 REQUIRED)
+    find_package(HDF5 REQUIRED COMPONENTS C)
```

Comment:
note: for all previous and from our project supported CMake versions the `C` component was chosen correctly. This is a backwards compatible fix for CMake 3.6+